### PR TITLE
Add information about publishing an extension

### PR DESCRIPTION
**Pull request by sypets**

### CODE REVIEW

File: Documentation/ExtensionArchitecture/PublishExtension/Index.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+.. _publish-extension:
+
+======================
+Publish Your Extension
+======================
+
+By publishing an extension, we mean making it publicly available. This may include
+all of or just a few of the following:
```
Comment (susannemoog) [inline Documentation/ExtensionArchitecture/PublishExtension/Index.rst]: I'd suggest to strongly recommend doing all of the following. Yes, sometimes there are strong reasons against one of these, but for a normal extension, all of that should be done.

File: Documentation/ExtensionArchitecture/PublishExtension/Index.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#. :ref:`Register your extension on extensions.typo3.org <publish-extension-ter>` (TER)
+#. Make your :ref:`extension source code available <publish-extension-git>`
+   on a public Git hoster such as GitHub,
+   Gitlab or Bitbucket
+#. :ref:`Register Your Extension for docs.typo3.org <publish-extension-docs>`:
+   This makes sure your extension documentation will be rendered on
+   docs.typo3.org
+
+
+We recommend to do all of these. Find more information in the following sections!
```
Comment (susannemoog) [inline Documentation/ExtensionArchitecture/PublishExtension/Index.rst]: do we need an exclamation mark?

File: Documentation/ExtensionArchitecture/PublishExtension/Index.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+====================================================
+
+
+See `Publish an Extension <https://extensions.typo3.org/faq/publish-an-extension/>`__
+for more information on how to publish an extension and check out the
+`FAQ <https://extensions.typo3.org/faq/>`__ as well.
+
+Advantages:
+
+* More visibility for your extension
```
Comment (susannemoog) [inline Documentation/ExtensionArchitecture/PublishExtension/Index.rst]: These advantages are mostly also (and maybe even more) true for packagist - maybe simply distinguish between availability in composer vs non-composer mode?

File: Documentation/ExtensionArchitecture/PublishExtension/Index.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+* Link to the source code (optional)
+
+
+.. _publish-extension-git:
+
+Publish Source Code on a Public Git Hosting Platform
+====================================================
+
+The TYPO3 community currently uses GitHub, GitLab and Atlassian Bitbucket to host
+the Git repository of their extensions.
```
Comment (susannemoog) [inline Documentation/ExtensionArchitecture/PublishExtension/Index.rst]: "repositories"

File: Documentation/ExtensionArchitecture/PublishExtension/Index.rst [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+Just add a repository to your workspace. Typically, the extension key is used
```
Comment (susannemoog) [inline Documentation/ExtensionArchitecture/PublishExtension/Index.rst]: Avoid using "Just" please.

File: Documentation/ExtensionArchitecture/PublishExtension/Index.rst (hunk 3 above)
Comment (sypets) [reply Documentation/ExtensionArchitecture/PublishExtension/Index.rst]: I now removed "More visibility for your extension" altogether. Is a little more terse and it does not make much sense to repeat that in most of the points.

**Pull request merged**